go/libraries/doltcore/sqle/history_table.go

```diff
@@ -250,12 +249,15 @@ func substituteWorkingHash(h hash.Hash, f []sql.Expression) []sql.Expression {
 	return ret
 }
 
-var historyTableCommitMetaCols = set.NewStrSet([]string{CommitHashCol, CommitDateCol, CommitterCol})
-
 func commitFilterForExprs(ctx *sql.Context, filters []sql.Expression) (doltdb.CommitFilter, error) {
 	filters = transformFilters(ctx, filters...)
 
-	return func(ctx context.Context, h hash.Hash, cm *doltdb.Commit) (filterOut bool, err error) {
+	return func(ctx context.Context, h hash.Hash, optCmt *doltdb.OptionalCommit) (filterOut bool, err error) {
+		cm, ok := optCmt.ToCommit()
+		if !ok {
+			return false, nil // NM4 TEST.
```

Comment:
Is the idea that we're *not* filtering this commit? Seems wrong. If it's not an error, we should be filtering it, right?